main.tf

```diff
@@ -44,6 +44,7 @@ resource "azapi_resource" "this" {
       hbiWorkspace        = var.hbi_workspace
       friendlyName        = "AMLManagedVirtualNetwork"
       keyVault            = local.key_vault_id
+      imageBuildCompute   = var.image_builder_compute_cluster_name
```

Comment:
Oh the validation of the image build compute only happens when you are actually opening a workspace not when creating it so this would not cause a loop.